python/singa/autograd.py

```diff
@@ -1945,3 +1945,29 @@ def backward(self, dy):
 
 def sub(a, b):
     return Sub()(a,b)[0]
+
+
+class Min(Operation):
+    def __init__(self):
+        super(Min, self).__init__()    
+    
+    def forward(self, a, b):
+        m = singa.__sub__(a,b)
+        mask0 = singa.LTFloat(m,0) 
+        mask00 = singa.__mul__(singa.LEFloat(m,0),a)
+        mask1 = singa.GTFloat(m,0)
+        mask11=singa.__mul__(mask1,b)
+        mask = singa.__add__(mask00,mask11)
+        
+        if training:
+            self.mask = mask
```

Comment:
mask is not used anymore?

Reply:
yes. only used in forward function

Reply:
then there is no need to store it in self.mask.

Reply:
yes, I just un-stored 'mask'